crates/sui-framework/packages/move-stdlib/sources/ascii.move

```diff
@@ -90,20 +120,44 @@ module std::ascii {
        bytes
     }
 
-    /// Unpack the `char` into its underlying byte.
+    /// Unpack the `char` into its underlying bytes.
     public fun byte(char: Char): u8 {
        let Char { byte } = char;
        byte
     }
 
-    /// Returns `true` if `b` is a valid ASCII character. Returns `false` otherwise.
+    /// Returns `true` if `b` is a valid ASCII character.
+    /// Returns `false` otherwise.
     public fun is_valid_char(b: u8): bool {
        b <= 0x7F
     }
 
-    /// Returns `true` if `byte` is an printable ASCII character. Returns `false` otherwise.
+    /// Returns `true` if `byte` is an printable ASCII character.
+    /// Returns `false` otherwise.
     public fun is_printable_char(byte: u8): bool {
        byte >= 0x20 && // Disallow metacharacters
        byte <= 0x7E // Don't allow DEL metacharacter
     }
+
+    /// Returns `true` if `string` is empty.
+    public fun is_empty(string: &String): bool {
+        string.bytes.is_empty()
+    }
+
+    /// Returns `true` if `string` is an alphanumeric ASCII string.
+    /// Returns `false` otherwise.
+    public fun is_alphanumeric(string: &String): bool {
```

Comment:
Hmmm. do we want to have this for strings, or do we think we will have some macro like `chars_all` with then doing `chars_all(|c| c.is_alphanumeric())`... unsure

Reply:
I like macro approach more, so let's sync about this and make a weighted call? This is not the only place where we could use macros!